CDRIVER-5733 fix behavior of `bson_string_truncate` (#1736)
mongodb/mongo-c-driver: src/libbson/doc/bson_string_truncate.rst

A ``\0`` byte will be placed where the end of the string occurs.

> .. warning:: This function will abort if the length of the resulting string (including the NULL terminator) would exceed ``UINT32_MAX``.

Reviewer: ```suggestion
.. warning:: The length of the resulting string (including the ``NULL`` terminator) MUST NOT exceed ``UINT32_MAX``.
```
Wording suggestion.

Author: Applied suggestion to other `bson_string_append*` functions with the same warning.
